src/mem/ruby/slicc_interface/AbstractCacheEntry.hh

```diff
@@ -68,23 +71,22 @@ class AbstractCacheEntry : public AbstractEntry
     void clearLocked();
     bool isLocked(int context) const;
 
-    void setSetIndex(uint32_t s) { m_set_index = s; }
-    uint32_t getSetIndex() const { return m_set_index; }
+    void setSetIndex(uint32_t s) { setPosition(s, getWay()); }
+    uint32_t getSetIndex() const { return getSet(); }
 
-    void setWayIndex(uint32_t s) { m_way_index = s; }
-    uint32_t getWayIndex() const { return m_way_index; }
+    void setWayIndex(uint32_t s) { setPosition(getSet(), s); }
+    uint32_t getWayIndex() const { return getWay() }
 
     // Address of this block, required by CacheMemory
     Addr m_Address;
     // Holds info whether the address is locked.
     // Required for implementing LL/SC operations.
     int m_locked;
 
-  private:
```

Comment:
Was m_Permission private in AbstractEntry.hh?  If so we should keep this line here.